src/lib.rs

```diff
@@ -84,6 +84,33 @@ async fn adjust_proxied_headers(req: &mut Request<Body>) -> Result<(), Error> {
     Ok(())
 }
 
+fn host_header_match_proxy_address(host_header: String, proxy_listener: &Listen) -> bool {
```

Comment:
The `Listen` struct should be replced by [SockerAddrV4](https://doc.rust-lang.org/std/net/struct.SocketAddrV4.html)

Reply:
Thanks, much easier!